File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> num_index; // érték : index
        for(int i = 0; i < nums.size(); i++){
            int to_find = target - nums[i];
            if(num_index.count(to_find)){
                return {num_index[to_find], i};
            }
            num_index[nums[i]] = i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> ValueIndex = new Dictionary<int, int>();

        for(int i = 0; i < nums.Count(); i++)
        {
            int toFind = target-nums[i];
            if(ValueIndex.ContainsKey(toFind))
            {
                return new int[] { i, ValueIndex[toFind] };
            }
            ValueIndex[nums[i]] = i;
        }
        return new int[0];
    }
}
